tools/Google.Cloud.Tools.ReleaseManager/ContainerCommands/GenerateCommand.cs

```diff
@@ -12,12 +12,65 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using Google.Cloud.Tools.Common;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Google.Cloud.Tools.ReleaseManager.ContainerCommands;
 
+/// <summary>
+/// Generates files for either a single API or all configured APIs. Expected options:
+/// - api-root: effectively the googleapis directory. We take a copy of this before running generation, unless we're running raw generation.
+/// - generator-input: optional, when omitted, run "raw generation"
+/// - output: root folder for result; required, must exist
+/// - api-path: e.g. google/cloud/functions/v2
+//    optional when generator-input is specified, when omitted, all configured APIs should be regenerated
+//    required when generator-input is not specified (because we don’t want to raw-generate everything in api-root)
+/// </summary>
 internal class GenerateCommand : IContainerCommand
 {
-    public int Execute(Dictionary<string, string> options) => throw new NotImplementedException();
+    public int Execute(Dictionary<string, string> options)
+    {
+        string apiRoot = options["api-root"];
+        string output = options["output"];
+        string generatorInput = options.GetValueOrDefault("generator-input");
+        string apiPath = options.GetValueOrDefault("api-path");
+
+        if (apiPath is null && generatorInput is null)
+        {
+            throw new UserErrorException("At least one of api-path or generator-input must be specified");
+        }
+
+        // Note: we expect the container to already have environment variables for
+        // protoc, protobuf tools root, the gRPC generator, and the GAPIC generator.
+        Environment.SetEnvironmentVariable(GenerateApisCommand.GeneratorInputDirectoryEnvironmentVariable, generatorInput);
+        Environment.SetEnvironmentVariable(GenerateApisCommand.GeneratorOutputDirectoryEnvironmentVariable, output);
+        Environment.SetEnvironmentVariable(GenerateApisCommand.GoogleApisDirectoryEnvironmentVariable, apiRoot);
+
+        var generatorCommand = new GenerateApisCommand();
+        if (generatorInput is null)
+        {
+            return generatorCommand.Execute(new[] { "--unconfigured", apiPath });
+        }
+        else
+        {
+            if (apiPath is not null)
+            {
+                var catalog = ApiCatalog.LoadFromGeneratorInput(generatorInput);
+                var id = catalog.Apis.FirstOrDefault(api => api.ProtoPath == apiPath)?.Id;
+                if (id is null)
+                {
+                    // TODO: Optionally, create a temporary configuration.
```

Comment:
Possibly using "oboard" command but without saving the result.

Reply:
Maybe. I'm really not sure at the moment, to be honest. It's possible that in-memory will be a better approach, so that we can make generator-input readonly...